.github/workflows/special-tests.yml

```diff
@@ -134,7 +134,7 @@ jobs:
       contents: read
       packages: read
     secrets: inherit
-    if: inputs.canary-run == 'true' || inputs.upgrade-to-newer-dependencies == 'true'
+    if: inputs.canary-run == 'true' || inputs.upgrade-to-newer-dependencies != 'false'
```

Comment:
Just for my understanding how is this different?

Reply:
It's different - because upgrade-to-newer-dependencies is either `false` or ... random hash :D. It's needed to make sure that appropriate Container layer gets invalidated (via build arg). I **think** I will need to add better description and likely change name of the parameter to reflect this because I keep on forgetting it myself.

Reply:
Actually.. i was wrong :) .

When I look in detail it looks like the inputs/outputs were the same "true/false" like everything else - only the --build-arg UPGRADE_TO_NEWER_DEPENDENCIES required the random value I mentioned above (due to docker layer caching). I made it clearer - I hope in https://github.com/apache/airflow/pull/38637 by renaming the build-arg to UPGRADE_TO_NEWER_DEPENDENCIES_HASH and correcting the descriptions.